docs/src/components/ButtonLink.tsx

```diff
@@ -0,0 +1,88 @@
+import React from "react";
+import styled, { css } from "styled-components";
+import { PolymorphicPropsWithoutRef } from "react-polymorphic-types";
+
+const DefaultComponent = "a";
+
+interface OwnProps {
+  type: "primary" | "secondary";
+  size?: "medium" | "large";
+  iconLeft?: React.ReactNode;
+  iconRight?: React.ReactNode;
+  children: React.ReactNode;
+}
+
+type Props<T extends React.ElementType = typeof DefaultComponent> = PolymorphicPropsWithoutRef<
+  OwnProps,
+  T
+>;
+
+const StyledComponent = styled(DefaultComponent)<Props>`
+  ${({ theme, type, size = "medium" }) => css`
```

Comment:
I think that way will be better, like we do in orbit usually

```
{
   const Component = asProp || DefaultComponent;
    return (
      <Component {...props}>
        {props.children}
      </Component>
    );
  },
)`
```

Reply:
Actually, I thnk this will not help me achieve my type checking goal, which is to allow the props of whichever component I pass to `as`, for example for Gatsby's `Link` i wanted the prop `to` to be available. So I don't want to inline the components to not cause any more complexity regarding that because I was barely able to achieve this 😅